### ISSUE TITLE
obj.Handle.pemCertificate

### ISSUE THREAD

**Issue by jdw2023** [mathworks-ref-arch/matlab-azure-services#2]
Line 17 in function clientSecretCredentialBuilder causes an error for me, when I am running credentials = configureCredentials("myCredentialConfiguration.json"), it gives:

K>> obj.Handle.pemCertificate(certificatePath)
Check for missing argument or incorrect argument data type in call to function 'pemCertificate'.

certificatePath is definitely a char and definitely exists - so I am not sure what is wrong.
Thanks

**Comment by maben-mw** [comment]
The issue is indeed probably not with the file itself, there are multiple checks in place to ensure the file is there before you can even reach the call which is throwing an error here.

The error appears to be caused by the fact that the whole `pemCertificate` methods simply does not exist *in the class on which we are trying to call it here* (regardless of the input, but since the input type *can* matter it is mentioned in the error message). 

Where the *real* issue here then appears that we are trying to work with the wrong class here. For certificate based client credentials we should likely be working with the Client*Certificate*CredentialBuilder whereas in practice we are accidentally working with Client*Secret*CredentialBuilder. I am not sure how this was overlooked but we will have to add this `ClientCertificateCredentialBuilder` class to the MATLAB interface and then update `configureCredentials` to use it where appropriate. (And add a few tests to properly test this).

I expect that we should be able to provide an updated version for this soon (likely somewhere next week).

**Comment by jdw2023** [comment]
Thanks for getting back to me, I look forward to trying it out when you have a fix.

**Comment by maben-mw** [comment]
I have just pushed version v0.3.1 which should fix the issue. Please download and try it.

For now closing the GitHub Issue, please reopen if any issues persist.

**Comment by jdw2023** [comment]
Thanks, yes that no longer crashes and seems to work. How do you get a token from the output of configureCredentials?

**Comment by maben-mw** [comment]
Typically when working with the `matlab-azure-services` package you do not explicitly obtain tokens by yourself; you configure the credentials object:

```matlab
c = configureCredentials('myConfig.json')
````

And then pass them to the Azure service client which you want to work with. E.g. if you want to work with Key Vault, you could then:

```matlab
secretClient = createKeyVaultClient('Type','Secret','VaultName','MyVault','Credentials',c)
```

The service client will then go and obtain the token whenever/wherever it needs it (which is likely then on the first interaction with that client). Typically that token is just used internally and not returned to the end-user.

If you do want/need to go and obtain tokens by yourself for usage outside of the service clients, you will first need to create a [TokenRequestContext](https://mathworks-ref-arch.github.io/matlab-azure-services/APIReference.html#azure-core-credential-tokenrequestcontext), configure it and then obtain the token, for example:

```matlab
% Configure credentials
c = configureCredentials('myConfig.json');
% Create a TokenRequestContext
tr = azure.core.credential.TokenRequestContext;
% Add scope to the context, '.default' is an example which typically *does*
% exist, but update as appropriate with the scopes you really require
tr = tr.addScopes('.default');
% Obtain a token using the context and credentials configuration
token = c.getToken(tr);
% Get the token as char array
t = token.getToken()
```

**Comment by jdw2023** [comment]
Thanks I will try that.
FYI I think that line 342 in configureCredentials.m should be builder = builder.authorityHost(settings.AuthorityHost);
instead of builder = builder.authorityHost(settings.authorityHost);

**Comment by maben-mw** [comment]
Thank you, yes you are absolutely correct,  that should be `AuthorityHost` this is now fixed in v0.3.2
